weaviate/templates/querierDeployment.yaml

```diff
@@ -0,0 +1,456 @@
+          - name: AWS_SECRET_ACCESS_KEY
+            valueFrom:
+              secretKeyRef:
+                name: weaviate-aws
+                key: AWS_SECRET_ACCESS_KEY
+            {{- end }}
+          {{- end  }}
+          {{- if index .Values "backups" "gcs" "enabled" }}
+            {{- if index .Values "backups" "gcs" "envconfig" }}
+              {{- range $key, $value := index .Values "backups" "gcs" "envconfig" }}
+          - name: {{ $key }}
+            value: {{ $value | quote }}
+              {{- end }}
+            {{- end }}
+            {{- if or (index .Values "backups" "gcs" "secrets") (index .Values "backups" "gcs" "envSecrets") }}
+              {{- if or (index .Values "backups" "gcs" "secrets" "GOOGLE_APPLICATION_CREDENTIALS") (index .Values "backups" "gcs" "envSecrets" "GOOGLE_APPLICATION_CREDENTIALS") }}
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: "/etc/gcp/credentials.json"
+              {{- end }}
+            {{- end }}
+          {{- end  }}
+          {{- if index .Values "backups" "azure" "enabled" }}
+            {{- if index .Values "backups" "azure" "envconfig" }}
+              {{- range $key, $value := index .Values "backups" "azure" "envconfig" }}
+          - name: {{ $key }}
+            value: {{ $value | quote }}
+              {{- end }}
+            {{- end }}
+            {{- if index .Values "backups" "azure" "envSecrets" }}
+              {{- range $key, $secret := $.Values.backups.azure.envSecrets }}
+          - name: {{ $key }}
+            valueFrom:
+              secretKeyRef:
+                name: {{ $secret }}
+                key: {{ $key | quote }}
+              {{- end }}
+            {{- end }}
+            {{- if and (index .Values "backups" "azure" "secrets") (index .Values "backups" "azure" "secrets" "AZURE_STORAGE_ACCOUNT") }}
+          - name: AZURE_STORAGE_ACCOUNT
+            valueFrom:
+              secretKeyRef:
+                name: backup-azure
+                key: AZURE_STORAGE_ACCOUNT
+            {{- end }}
+            {{- if and (index .Values "backups" "azure" "secrets") (index .Values "backups" "azure" "secrets" "AZURE_STORAGE_KEY") }}
+          - name: AZURE_STORAGE_KEY
+            valueFrom:
+              secretKeyRef:
+                name: backup-azure
+                key: AZURE_STORAGE_KEY
+            {{- end }}
+            {{- if and (index .Values "backups" "azure" "secrets") (index .Values "backups" "azure" "secrets" "AZURE_STORAGE_CONNECTION_STRING") }}
+          - name: AZURE_STORAGE_CONNECTION_STRING
+            valueFrom:
+              secretKeyRef:
+                name: backup-azure
+                key: AZURE_STORAGE_CONNECTION_STRING
+            {{- end }}
+          {{- end  }}
+          - name: CLUSTER_JOIN
+            value: {{ .Values.service.name }}-headless.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
```

Comment:
shouldn't we use a separate headless service for `weaviate-querier` deployment? how about scaling this service? and the raft response from this service? should querier be aware of all Weaviate pods?